### PR TITLE
feat : auth and history

### PR DESCRIPTION
**Pull request by ragini07**
1.Added authentication (login , signup , logout)
2.Add to history
Login using admin cred and on click on video card it will be added to history
3.Delete single video from history
4.clear history

https://deploy-preview-2--fitsuvideolibrary.netlify.app/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
         <Route path='/login' element={<Login />} />
         <Route path='/signup' element={<SignUp />} />
         <Route path='/videos' element={<Videos/>} />
         <Route path='/video/:id' element={<VideoDetail/>} />
         <Route path='/profile' element={<UserProfile/>}/>
+        <Route path='/history' element={<PrivateRoute><History/></PrivateRoute>} />
+        <Route path='/likedvideos' element={<PrivateRoute><Like/></PrivateRoute>} />
+        <Route path='/watchlater' element={<PrivateRoute><WatchLater/></PrivateRoute>} />
+        <Route path='/playlists' element={<PrivateRoute><PlayLists/></PrivateRoute>} />
+        <Route path='/playlist/:id' element={<PrivateRoute><SinglePlayList/></PrivateRoute>} />
```
Comment (znjs) [inline src/App.js:22]: can use `<Outlet/>` in react-router-dom

File: src/Components/Auth/Login.js [hunk 2]
```diff
@@ -1,9 +1,68 @@
...
+import {useState} from 'react'
+import {useAuth} from '../../Context/auth-context' 
+import {useNavigate , Link , Navigate} from 'react-router-dom'
+import {SideBar} from '../SideBar/SideBar'
 function Login() {
-  return (
-    <div>Login</div>
+
+  const [email,setEmail] = useState("")
+  const [password,setPassword] = useState("")
```
Comment (znjs) [inline src/Components/Auth/Login.js:9]: can combine into a single `useState`

File: src/Components/Auth/auth.css [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    justify-content: center;
+    align-items: center;
+    height: 100%;
+    text-align: center;
+   
+}
+.span-txt{
+    line-height: 20px;
+    font-size: 14px;
+    margin:10px;
```
Comment (znjs) [inline src/Components/Auth/auth.css]: use relative units if possible

File: src/Components/Videos/VideoCard.js [hunk 4]
```diff
@@ -1,18 +1,88 @@
-import React from 'react'
+import {useState} from 'react'
 import {useNavigate} from 'react-router-dom'
+import {useAuth} from '../../Context/auth-context'
+import {useUser} from '../../Context/user-context'
+import {useVideos} from '../../Context/data-context'
```
Comment (znjs) [inline src/Components/Videos/VideoCard.js:5]: can use `index` to export (barrel export)

File: src/Components/Videos/Videos.css [hunk 5]
```diff
@@ -24,8 +24,28 @@
...
 }
 .card-options{
     padding-right: 0.4rem;
+}
+.list-item{
+    position: absolute;
+    right: 0.8rem;
+    top: 1.1rem;
+    max-width: 14rem;
+    /* border: 1px solid var(--primary-color); */
```
Comment (znjs) [inline src/Components/Videos/Videos.css]: can remove the commented code

File: src/Service/userAction.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+}
+export const removeFromHistory = async(dispatchUserData , token , video) => {
+    try{
+       const {data , status } = await axios.delete(`/api/user/history/${video._id}`,{
+            headers : {
+                authorization : token
+            }
+        })
+        if(status === 200){
+            dispatchUserData({type : "REMOVE_FROM_HISTORY" , payload : video})
```
Comment (znjs) [inline src/Service/userAction.js:41]: can use variables to avoid typo bugs

File: src/Service/userAction.js [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+       const {data , status} = await axios.post('/api/user/likes',{
+            video
+        },{
+            headers : {
+                authorization : token
+            }
+        })
+        console.log("add to liked videos" , data , status)
+        if(status === 200 || status === 201)
+          dispatchUserData({type : "ADD_TO_LIKED_VIDEOS" , payload : video})
```
Comment (znjs) [inline src/Service/userAction.js:71]: can use constants instead of strings

File: src/Service/userAction.js [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+export const addToPlayList = async (dispatchUserData , token , playlist , video) => {
+    try {
+       const {data , status} = await axios.post(`/api/user/playlists/${playlist._id}`,{
+            video
+        },{
+            headers : {
+                authorization : token
+            }
+        })
+        console.log("add to playlist" , data , status)
```
Comment (deepak29-git) [inline src/Service/userAction.js]: remove console

File: src/Service/userAction.js [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+export const addToWatchLater = async (dispatchUserData , token , video) => {
+    try {
+       const {data , status} = await axios.post('/api/user/watchlater',{
+            video
+        },{
+            headers : {
+                authorization : token
+            }
+        })
+        console.log("add to watch later" , data , status)
```
Comment (deepak29-git) [inline src/Service/userAction.js]: remove console

**Review by deepak29-git — APPROVED** [review]

**Review by znjs — APPROVED** [review]
- Good folder structure
- Good naming conventions used

File: src/App.js (hunk 1 above)
Comment (ragini07) [reply src/App.js:22]: Thanks for suggestion . I am not aware of what it does. Will read and implement

File: src/Service/userAction.js (hunk 6 above)
Comment (ragini07) [reply src/Service/userAction.js:41]: I will at end make all reducer action type constants

**Pull request merged**